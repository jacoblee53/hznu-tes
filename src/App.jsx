import React from 'react'
import { HashRouter as Router, Route, Switch } from 'react-router-dom'
import { inject, observer } from 'mobx-react'

import Dashboard from './component/Dashboard'
import * as urls from './constant/urls.js'
import jwt from './util/token'
import Loadable from './component/Loadable'

@inject('userActions')
@observer
class App extends React.Component {
  constructor(props) {
    super(props)

    if (jwt.getToken()) {
      props.userActions.autoLogin()
    } else {
      window.location.assign(
        location.origin + location.pathname + '#' + '/login'
      )
    }
  }

  render() {
    return (
      <Router>
        <Switch>
          <Route
            exact
            path={urls.LOGIN}
            component={Loadable({ loader: () => import('./app/login')})}
          />
          <Route
            path='/'
            render={() => (
              <div className='app-root'>
                <Dashboard>
                  <Switch>
                    <Route exact path={urls.HOME} />
                    <Route
                      exact
                      path={urls.TASK_MANAGE}
                      component={Loadable({ loader: () => import('./app/task-manage')})}
                    />
                    <Route
                      exact
                      path={urls.EVALUATE}
                      component={Loadable({ loader: () => import('./app/evaluate')})}
                    />
                    <Route
                      exact
                      path={urls.SCORE_MANAGE}
                      component={Loadable({ loader: () => import('./app/score-manage')})}
                    />
                    <Route
                      exact
                      path={urls.STANDARD_MANAGE}
                      component={Loadable({ loader: () => import('./app/standard-manage')})}
                    />
                    <Route
                      exact
                      path={urls.CLASS_MANAGE}
                      component={Loadable({ loader: () => import('./app/class-manage')})}
                    />
                  </Switch>
                </Dashboard>
              </div>
            )}
          />
        </Switch>
      </Router>
    )
  }
}

export default App
