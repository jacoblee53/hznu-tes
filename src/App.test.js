import React from 'react'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'
import injects from './inject'
import App from './app'

let wrappedComponent = <App {...injects} />

let wrapper

describe('src/app.test.js', () => {
  it('should render correctly', () => {
    wrapper = mount(wrappedComponent)
    expect(wrapper.html()).toBeTruthy()
  })

  it('should matchSnapshot', () => {
    const tree = renderer.create(wrappedComponent)
    expect(tree).toMatchSnapshot()
  })
})
