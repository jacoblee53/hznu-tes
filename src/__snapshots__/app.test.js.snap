// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/app.test.js should matchSnapshot 1`] = `
<div
  style={
    Object {
      "height": "100vh",
      "width": "100%",
    }
  }
>
  <div
    style={
      Object {
        "WebkitBoxAlign": "center",
        "alignItems": "center",
        "display": "flex",
        "height": "100%",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <div
      className="sk-spinner line-scale-pulse-out-rapid"
      color="#5282ef"
      style={
        Object {
          "color": "#5282ef",
        }
      }
    >
      <div />
      <div />
      <div />
      <div />
      <div />
    </div>
  </div>
</div>
`;
